File: Python/Convert Sorted Array to Binary Search Tree.py
class Solution:
    # @param num, a list of integers
    # @return a tree node
    def sortedArrayToBST(self, num):
        if not num:
            return None
        length = len(num)
        node = TreeNode(num[length//2])
        node.left = self.sortedArrayToBST(num[:length//2] if length//2 else None)
        node.right = self.sortedArrayToBST(num[length // 2 +1 :] if length //2 < length-1 else None)
        
        return node

File: Java/src/Convert Sorted Array to Binary Search Tree.java
 public class Solution {
        public TreeNode sortedArrayToBST(int[] num) {
            return helper(num,0,num.length-1);
        }
        private TreeNode helper(int[] num, int start,int end)
        {
            if(start>end) return null;
            int middle = (start+end)/2;
            TreeNode node = new TreeNode(num[middle]);
            node.left = helper(num,start,middle-1);
            node.right = helper(num,middle+1,end);
            return node;
        }
}